internal/core/metadata/operators/device_service/update.go

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package device_service
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/errors"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type UpdateOpStateByIdExecutor interface {
```

Comment:
Would it be better to create one exported executor interface for all the updating functionality, since they all accept 0 arguments and return 1 error? 

We are doing this in the `device_profile` get operator:
https://github.com/edgexfoundry/edgex-go/blob/8627af8819795f333cf7a38c6bcd9bb746df233b/internal/core/metadata/operators/device_profile/get.go#L16-L19

In this case we have two types of interfaces as some of the getters return one `device_profile` while others return a slice of `device_profile` so we created 2 interfaces which cover all of the 7 getters.

Here is the interface for the executors which return more a slice:
https://github.com/edgexfoundry/edgex-go/blob/8627af8819795f333cf7a38c6bcd9bb746df233b/internal/core/metadata/operators/device_profile/get.go#L11-L14